src/core/dirlistjob.cpp

```diff
@@ -62,6 +63,12 @@ void DirListJob::exec() {
             false
     };
     if(enu) {
+        bool bindCutFiles = false;
+        if(cutFilesHashSet_
+                && cutFilesHashSet_->size() < 450 /* give up if more than 450 cut files */) {
```

Comment:
It looks like it's OK to remove it, it's fast for >1000 files, but be aware it's very different code path when selecting & cutting files inside a directory and go to a directory with already selected & cut files - hash set search is done only in the latter case.